### PR TITLE
feat: record changes during role mapping

### PR DESCRIPTION
**Pull request by voikin**
add separated record builder to record events for each user

_more details in `ASP-1227`_

### CODE REVIEW

File: controllers/ldap.go [hunk 1]
```diff
@@ -422,6 +422,10 @@ func (c *ApiController) SyncLdapUsers() {
 		c.ResponseError(err.Error())
 		return
 	}
+
+	mappingRb := object.NewRecordBuilderFromCtx(c.Ctx)
+	mappingCtx := context.WithValue(goCtx, object.RoleMappingRecordDataKey, mappingRb)
```
Comment (svgrafov) [inline controllers/ldap.go:427]: If you dont need to separately store mappingCtx and goCtx, you can rewrite goCtx value with result of context.WithValue(), resulting in
`
goCtx := context.WithValue(goCtx, object.RoleMappingRecordDataKey, mappingRb)
`
This makes current scope less cluttered with not needed variables.

File: object/role.go [hunk 2]
```diff
@@ -636,5 +637,44 @@ func SyncRolesToUser(ctx context.Context, user *User, roleIds []string) error {
...
+	newUserRoles, err := getRolesByUserInternal(userId)
+	if err != nil {
+		return err
+	}
+
+	recordUserMappedRoles(ctx, user.GetId(), currentUserRoles, newUserRoles)
 	return nil
 }
+
+func recordUserMappedRoles(ctx context.Context, userID string, oldRoles, newRoles []*Role) {
```
Comment (svgrafov) [inline object/role.go:650]: please add test for this

**Review by dplynskiy — APPROVED** [review]

**Review by svgrafov — CHANGES_REQUESTED** [review]
Conflicts with my PR. Should be merged before mine and I will fix conflicts myself

File: controllers/ldap.go (hunk 1 above)
Comment (voikin) [reply controllers/ldap.go]: done

File: object/role.go (hunk 2 above)
Comment (voikin) [reply object/role.go]: done

**Review by svgrafov — APPROVED** [review]

**Pull request merged**